packages/eslint-plugin/src/rules/no-empty-interface.ts

```diff
@@ -76,26 +77,41 @@ export default util.createRule<Options, MessageIds>({
 
             // Check if interface is within ambient declaration
             let useAutoFix = true;
+            let hasSuggest = true;
+            const scope = context.getScope();
+
             if (util.isDefinitionFile(filename)) {
-              const scope = context.getScope();
               if (scope.type === 'tsModule' && scope.block.declare) {
```

Comment:
[Nit] Now that the `const scope = ...` line is earlier, these two `if` statements can be combined:

```ts
if (
  util.isDefinitionFile(filename) &&
  scope.type === "tsModule" &&
  scope.block.declare
) {
  useAutoFix = false;
}
```

...so the code can be simplified a bit more, to give `useAutoFix` an initial value:

```ts
let useAutoFix = ...;
```

I would even suggest making a separate variable for whether it's merged with a class declaration:

```ts
const mergedWithClassDeclaration = scope.set
  .get(node.id.name)
  ?.defs?.some((def) => def.node.type === AST_NODE_TYPES.ClassDeclaration);

const useAutoFix =
  !mergedWithClassDeclaration && ...;
```

Thoughts?